crosscluster/physical: TestAlterTenantCompleteToLatest failed

crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://tanzanite.cluster.engflow.com/invocations/default/9d17ad83-65dd-4b6e-a688-e4453de0cb36?testReportRun=1&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [cc3365feadf12f4669a164f8033ad3ddf1f80fc3](https://github.com/cockroachdb/cockroach/commits/cc3365feadf12f4669a164f8033ad3ddf1f80fc3):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAlterTenantCompleteToLatest2613121574
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:89: error executing query="\n\tCREATE DATABASE d;\n\tCREATE TABLE d.t1(i int primary key, a string, b string);\n\tCREATE TABLE d.t2(i int primary key);\n\tINSERT INTO d.t1 (i) VALUES (42);\n\tINSERT INTO d.t2 VALUES (2);\n\tUPDATE d.t1 SET b = 'world' WHERE i = 42;\n\t" args=[]: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/106/1/42/0): "sql txn" meta={id=6b9b5671 key=/Tenant/10/Table/3/1/107/2/1 iso=Serializable pri=0.01184023 epo=0 ts=1738332767.536423981,2 min=1738332767.487142971,0 seq=0} lock=true stat=PENDING rts=1738332767.487142971,0 wto=false gul=1738332767.987142971,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterTenantCompleteToLatest2613121574
I250131 14:12:47.762678 8371 13@kv/kvserver/replica_command.go:481  [T1,Vsystem,n1,split,s1,r86/1:/Tenant/1{0/Table/…-1}] 2  initiating a split of this range at key /Tenant/10/Table/106 [r87] (span config)
I250131 14:12:47.773710 517 13@spanconfig/spanconfigstore/store.go:364  [T1,Vsystem,n1] 3  changing the spanconfig for span:/Table/4{2-3} from:{RangeMinBytes:134217728 RangeMaxBytes:536870912 GCPolicy:{TTLSeconds:14400 ProtectionPolicies:[] IgnoreStrictEnforcement:true} GlobalReads:false NumReplicas:5 NumVoters:0 Constraints:[] VoterConstraints:[] LeasePreferences:[] RangefeedEnabled:true ExcludeDataFromBackup:false} to:{RangeMinBytes:134217728 RangeMaxBytes:536870912 GCPolicy:{TTLSeconds:3600 ProtectionPolicies:[] IgnoreStrictEnforcement:true} GlobalReads:false NumReplicas:5 NumVoters:0 Constraints:[] VoterConstraints:[] LeasePreferences:[] RangefeedEnabled:true ExcludeDataFromBackup:false}
I250131 14:12:47.775195 517 13@spanconfig/spanconfigstore/store.go:364  [T1,Vsystem,n1] 4  changing the spanconfig for span:/Table/4{3-4} from:{RangeMinBytes:134217728 RangeMaxBytes:536870912 GCPolicy:{TTLSeconds:14400 ProtectionPolicies:[] IgnoreStrictEnforcement:true} GlobalReads:false NumReplicas:5 NumVoters:0 Constraints:[] VoterConstraints:[] LeasePreferences:[] RangefeedEnabled:true ExcludeDataFromBackup:false} to:{RangeMinBytes:134217728 RangeMaxBytes:536870912 GCPolicy:{TTLSeconds:3600 ProtectionPolicies:[] IgnoreStrictEnforcement:true} GlobalReads:false NumReplicas:5 NumVoters:0 Constraints:[] VoterConstraints:[] LeasePreferences:[] RangefeedEnabled:true ExcludeDataFromBackup:false}
I250131 14:12:47.776602 517 13@spanconfig/spanconfigstore/store.go:364  [T1,Vsystem,n1] 5  changing the spanconfig for span:/Table/6{1-2} from:{RangeMinBytes:134217728 RangeMaxBytes:536870912 GCPolicy:{TTLSeconds:14400 ProtectionPolicies:[] IgnoreStrictEnforcement:true} GlobalReads:false NumReplicas:5 NumVoters:0 Constraints:[] VoterConstraints:[] LeasePreferences:[] RangefeedEnabled:true ExcludeDataFromBackup:false} to:{RangeMinBytes:134217728 RangeMaxBytes:536870912 GCPolicy:{TTLSeconds:3600 ProtectionPolicies:[] IgnoreStrictEnforcement:true} GlobalReads:false NumReplicas:5 NumVoters:0 Constraints:[] VoterConstraints:[] LeasePreferences:[] RangefeedEnabled:true ExcludeDataFromBackup:false}
I250131 14:12:47.782385 8371 13@kv/kvserver/batcheval/cmd_end_transaction.go:1289  [T1,Vsystem,n1,s1,r86/1:/Tenant/1{0/Table/…-1}] 6  falling back to accurate stats computation because the in-split LHS or RHS is empty
I250131 14:12:47.834242 8408 13@kv/kvserver/replicate_queue.go:783  [T1,Vsystem,n1,replicate,s1,r87/1:/Tenant/1{0/Table/…-1}] 7  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
--- FAIL: TestAlterTenantCompleteToLatest (10.37s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47040

crosscluster/physical.TestAlterTenantCompleteToLatest [failed](https://tanzanite.cluster.engflow.com/invocations/default/0cb4b4ee-e604-43ff-b1b1-e262d47c5916?testReportRun=2&testReportShard=2&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL3BoeXNpY2FsOnBoeXNpY2FsX3Rlc3Q=) on master @ [f7d019d5e675fa2711b61ffd54a9990b6a1a1da8](https://github.com/cockroachdb/cockroach/commits/f7d019d5e675fa2711b61ffd54a9990b6a1a1da8):


```
=== RUN   TestAlterTenantCompleteToLatest
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestAlterTenantCompleteToLatest1736481177
    test_log_scope.go:76: use -show-logs to present logs inline
    testutils.go:89: error executing query="\n\tCREATE DATABASE d;\n\tCREATE TABLE d.t1(i int primary key, a string, b string);\n\tCREATE TABLE d.t2(i int primary key);\n\tINSERT INTO d.t1 (i) VALUES (42);\n\tINSERT INTO d.t2 VALUES (2);\n\tUPDATE d.t1 SET b = 'world' WHERE i = 42;\n\t" args=[]: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_ASYNC_WRITE_FAILURE - missing intent on: /Tenant/10/Table/106/1/42/0): "sql txn" meta={id=763c45de key=/Tenant/10/NamespaceTable/30/1/104/105/"t2"/4/1 iso=Serializable pri=0.04078378 epo=0 ts=1738415281.352877517,2 min=1738415281.225966166,0 seq=0} lock=true stat=PENDING rts=1738415281.225966166,0 wto=false gul=1738415281.725966166,0
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterTenantCompleteToLatest1736481177
I250201 13:08:01.565345 7690 13@kv/kvserver/replica_command.go:481  [T1,Vsystem,n1,split,s1,r86/1:/Tenant/1{0/Table/…-1}] 2  initiating a split of this range at key /Tenant/10/Table/106 [r87] (span config)
I250201 13:08:01.584688 7690 13@kv/kvserver/batcheval/cmd_end_transaction.go:1289  [T1,Vsystem,n1,s1,r86/1:/Tenant/1{0/Table/…-1}] 3  falling back to accurate stats computation because there are no pre-split LHS stats (or they're empty)
I250201 13:08:01.606084 7718 13@kv/kvserver/replicate_queue.go:783  [T1,Vsystem,n1,replicate,s1,r87/1:/Tenant/1{0/Table/…-1}] 4  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
--- FAIL: TestAlterTenantCompleteToLatest (9.75s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=2</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTenantCompleteToLatest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/crosscluster/replicationtestutils/testutils.go
+++ b/pkg/crosscluster/replicationtestutils/testutils.go
@@ -86,13 +86,19 @@ var DefaultTenantStreamingClustersArgs = TenantStreamingClustersArgs{
 	SrcTenantName: roachpb.TenantName("source"),
 	SrcTenantID:   roachpb.MustMakeTenantID(10),
 	SrcInitFunc: func(t *testing.T, sysSQL *sqlutils.SQLRunner, tenantSQL *sqlutils.SQLRunner) {
+		// Disable autocommit_before_ddl in order to reduce the amount of overhead
+		// during the setup. We've seen that this step can flake under race builds
+		// due to overload.
 		tenantSQL.Exec(t, `
+	BEGIN;
+	SET LOCAL autocommit_before_ddl = false;
 	CREATE DATABASE d;
 	CREATE TABLE d.t1(i int primary key, a string, b string);
 	CREATE TABLE d.t2(i int primary key);
 	INSERT INTO d.t1 (i) VALUES (42);
 	INSERT INTO d.t2 VALUES (2);
 	UPDATE d.t1 SET b = 'world' WHERE i = 42;
+	COMMIT;
 	`)
 	},
 	SrcNumNodes:         1,

